src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -3073,22 +3090,21 @@ private void visitFunctionParameters(
       Iterator<Node> parameters = ftype.getParameters().iterator();
       Iterator<String> names = null;
       Node functionSource = ftype.getSource();
-      if (functionSource != null) {
+      if (functionSource != null && functionSource.isClass()) {
```

Comment:
Question for John, is there or going to be a 'ClassType'? This function is odd, because we use it both for functions and classes, but maybe classes should have been handled separately altogether.